sdk/src/assertions/bmff_hash.rs

```diff
@@ -224,45 +325,447 @@ impl BmffHash {
         }
     }
 
-    pub fn verify_in_memory_hash(&self, data: &[u8], alg: Option<String>) -> Result<()> {
+    pub fn verify_in_memory_hash(
+        &self,
+        data: &[u8],
+        alg: Option<&str>,
+    ) -> crate::error::Result<()> {
+        let mut reader = Cursor::new(data);
+
+        self.verify_stream(&mut reader, alg)
+    }
+
+    // The BMFFMerklMaps are stored contiguous in the file.  Break this Vec into groups based on
+    // the MerkleMap it matches.
+    fn split_bmff_merkle_map(
+        &self,
+        bmff_merkle_map: Vec<BmffMerkleMap>,
+    ) -> crate::Result<HashMap<u32, Vec<BmffMerkleMap>>> {
+        let mut current = bmff_merkle_map;
+        let mut output = HashMap::new();
+        if let Some(mm) = self.merkle() {
+            for m in mm {
+                let rest = current.split_off(m.count as usize);
+
+                if current.len() == m.count as usize {
+                    output.insert(m.local_id, current.to_owned());
+                } else {
+                    return Err(Error::HashMismatch("MerkleMap count incorrect".to_string()));
+                }
+                current = rest;
+            }
+        } else {
+            output.insert(0, current);
+        }
+        Ok(output)
+    }
+
+    pub fn verify_hash(&self, asset_path: &Path, alg: Option<&str>) -> crate::error::Result<()> {
```

Comment:
why don't we need a file_io feature on this?

Reply:
We could but it is only called from store functions that are already behind that flag